nixos/modules/virtualisation/nixos-containers.nix

```diff
@@ -173,6 +173,9 @@ let
         ${optionalString (cfg.additionalCapabilities != null && cfg.additionalCapabilities != [])
           ''--capability="${concatStringsSep "," cfg.additionalCapabilities}"''
         } \
+        ${optionalString (cfg.systemCallFilter != null && cfg.systemCallFilter != [])
```

Comment:
I see this has been done by the other options, but it doesn't make any sense to me. How could `cfg.systemCallFilter` ever be `null` if it has already been typechecked with the type `listOf str` and `[ ]` as default?

Reply:
I fully agree with you. I just copied this check from the line just above about `additionalCapabilities`, which has the same type and the same default value. I figured there must be something I didn't understand about validation in Nix. Could you just confirm that this check if unnecessary?

Reply:
This check seems to go back to the creation of the `additionalCapabilities` option (0d1e1b1810081f8fb6496e2879bfa8326d79e580). However, even at that time, the option already had the same type and default. This is really a mystery for me.

Reply:
Feel free to just not do it. I really don't believe there's any good reason to. Maybe even make another commit, fixing up the others if you'd like. We have quite a large test suite for these (see `nixos/tests/containers-*.nix`), so it probably shouldn't be a problem if all of them succeed.